topydo/ui/TodoListWidget.py

```diff
@@ -102,9 +104,41 @@ def keypress(self, p_size, p_key):
             # make sure to accept normal shortcuts again
             self.keystate = None
             return
+        elif self.keystate in ['p', 'ps']:
+            if p_key not in ['d', 'w', 'm', 'y']:
+                if p_key.isdigit():
+                    self._pp_offset += p_key
+                elif self.keystate == 'p' and p_key == 's':
+                    self.keystate = 'ps'
+                else:
+                    self._pp_offset = ''
+                    self.keystate = None
+            else:
+                self._postpone_selected_item(p_key)
+                self._pp_offset = ''
+                self.keystate = None
+
+            return
+        elif self.keystate == 'r':
+            if p_key.isalpha():
+                self._pri_selected_item(p_key)
+            self.keystate = None
+            return
 
         if p_key == 'x':
             self._complete_selected_item()
+        elif p_key == 'p':
+            self.keystate = 'p'
+        elif p_key == 'd':
+            self._remove_selected_item()
+        elif p_key == 'e':
+            self._edit_selected_item()
+            # force screen redraw after editing
+            return self.listbox.keypress(p_size, 'ctrl l')
```

Comment:
This looks hacky. I would emit a _refresh_ signal here, caught inside Main.py which can call `self.mainloop.draw_screen()`.